### PR TITLE
OAK-9767 - Support Ints / Longs

### PR DESCRIPTION
**Pull request by klcodanr**
Adding support for setting the field type when retrieving terms using the LuceneIndexMBean

### CODE REVIEW

File: oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBean.java [hunk 1]
```diff
@@ -75,6 +75,23 @@ String[] getFieldTermsInfo(
...
+            "This allows to closely investigate what is stored in the index.")
+    String[] getFieldTermsInfo(
+            @Name("indexPath")
+            @Description("The index path (empty for all indexes)")
+                    String indexPath,
+            @Name("field")
+            @Description("The field name (empty for all fields)")
+                    String field,
+            @Name("fieldType")
+            @Description("The type of the field (empty for string)")
```
Comment (thomasmueller) [inline oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBean.java:88]: Could we add examples in the documentation (java.lang.Integer, int,...)?

File: oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImpl.java [hunk 2]
```diff
@@ -46,6 +47,7 @@
 import org.apache.jackrabbit.oak.api.CommitFailedException;
 import org.apache.jackrabbit.oak.api.jmx.Name;
 import org.apache.jackrabbit.oak.commons.PathUtils;
+import org.apache.jackrabbit.oak.commons.StringUtils;
```
Comment (fabriziofortino) [inline oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImpl.java]: unused import
```suggestion
```

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.jackrabbit.oak.plugins.index.lucene;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java]: unused
```suggestion
```

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        LuceneIndexNode indexNode = mock(LuceneIndexNode.class);
+        when(indexNode.getSearcher()).thenAnswer(inv -> new IndexSearcher(DirectoryReader.open(directory)));
+        indexes.put(path, indexNode);
+        return writer;
+    }
+
+    private void assertTermsMatch(String expectedFile, String[] actualTermsResult) throws IOException {
+        String[] expectedtermsResult = IOUtils
+                .readLines(getClass().getResourceAsStream(expectedFile), StandardCharsets.UTF_8).toArray(new String[0]);
+        assertArrayEquals(expectedtermsResult, actualTermsResult);
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java]: fix camel case
```suggestion
        String[] expectedTermsResult = IOUtils
                .readLines(getClass().getResourceAsStream(expectedFile), StandardCharsets.UTF_8).toArray(new String[0]);
        assertArrayEquals(expectedTermsResult, actualTermsResult);
```

File: oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+
+    @Test
+    public void testGetFieldTermsInfo_withType() throws IOException {
+        String INDEX_PATH = "/oak:index/test-index";
+        IndexWriter indexWriter = addNodeIndex(INDEX_PATH);
+        for (int i = 0; i < 10; i++) {
+            Document doc = new Document();
+            doc.add(new StringField("string", "value-" + i, Store.NO));
+            doc.add(new LongField("long", (long) i, Store.NO));
```
Comment (fabriziofortino) [inline oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneIndexMBeanImplTest.java:91]: redundant cast
```suggestion
            doc.add(new LongField("long", i, Store.NO));
```

**Review by fabriziofortino — APPROVED** [review]
LGTM

**Review by thomasmueller — APPROVED** [review]

**Comment by thomasmueller** [comment]
Merged on 2022-06-22

**Pull request merged**